### ISSUE TITLE
ParameterValue rejects JAX arrays/tracers, blocking engine-level jax.vmap

### ISSUE THREAD

**Issue by jc-macdonald** [ACCIDDA/flepimop2#269]
## Environment

- OS: macOS 15 (Darwin 24)
- Python version: 3.13.0
- `flepimop2` version (or commit, if developing from source): `0.3.0.dev0` (commit `8b54526`)

## Reproducible example

```python
import jax
import jax.numpy as jnp
import numpy as np
from flepimop2.axis import ResolvedShape
from flepimop2.parameter.abc import ParameterValue

shape = ResolvedShape(axis_names=("draw",), sizes=(3,))

# (1) Concrete JAX array — silently round-trips through host memory.
pv = ParameterValue(value=jnp.asarray([1.0, 2.0, 3.0]), shape=shape)
print(type(pv.value).__module__)  # 'numpy'  -- the JAX-ness is lost

# (2) JAX tracer (e.g. inside jax.vmap) — hard error.
def make_pv(x):
    return ParameterValue(value=jnp.broadcast_to(x, (3,)), shape=shape)

jax.vmap(make_pv)(jnp.asarray([1.0, 2.0, 3.0]))
# jax.errors.TracerArrayConversionError:
# The numpy.ndarray conversion method __array__() was called on traced array
# with shape float32[3]
```

The offending line is in `src/flepimop2/parameter/abc/__init__.py::ParameterValue.__post_init__`:

```python
value = np.asarray(self.value, dtype=np.float64)
```

This is invoked unconditionally, so any JAX tracer that flows through a `ParameterABC` subclass — e.g. when an inference / prior-predictive driver wraps the engine in `jax.vmap` over a sample dimension — gets stuck at the `ParameterValue` boundary.

## Brief explanation

`ParameterValue` is the contract between user `ParameterABC` plugins and the engine. Today it is hard-pinned to a NumPy `Float64NDArray`, but the `flepimop2-op-engine` runner is a JAX function and ABC plugins can legitimately want to produce JAX-typed output:

- Concrete `jax.Array` already works *by accident* (host round-trip) but loses the device/dtype/PyTree identity.
- `jax.vmap` / `jax.jit` tracers fail outright with `TracerArrayConversionError`, which blocks driving an existing `engine.run(...)` call with `jax.vmap` over a draw axis (vectorised prior-predictive / NUTS sampling).

A straightforward, JAX-optional fix:

1. Detect array-like values whose module is `jax` / `jaxlib` and **skip** the `np.asarray` cast (use the value as-is, validate `.shape`).
2. Otherwise keep the existing `np.asarray(..., dtype=np.float64)` path, preserving bit-exact behaviour for every existing NumPy-only flow.
3. Loosen the `value: Float64NDArray` annotation to `Float64NDArray | jax.Array` (using a `TYPE_CHECKING` import so jax stays an optional dependency at runtime).

This keeps `flepimop2` jax-free at runtime (the check is duck-typed via `type(value).__module__`), preserves all existing tests, and unblocks the engine-level `jax.vmap` use case for downstream inference back-ends.

Happy to send a PR — it’s a ~10-line patch plus tests.

### Concrete downstream use case

In ACCIDDA/flepimop2-demo we have a `numpyro_predictive` process plugin that draws N prior samples via `numpyro.infer.Predictive`, then injects them into a parametric expression via a `ContextVar`-based substitution context. Substituted values can be JAX scalars / tracers, and the parametric plugin already has a dual-path numpy ↔ jax.numpy backend selector. The only place that still forces NumPy is `ParameterValue.__post_init__`. With this fix, the per-draw serial loop becomes a single `jax.vmap(run_one)(prior_samples)` call against the existing JIT'd diffrax engine.

**Comment by jc-macdonald** [comment]
Closing — handling this entirely on the downstream side (ACCIDDA/flepimop2-demo) without requiring an upstream change.

The dual-path approach in the demo's parametric / distributions plugins works against the unmodified upstream `ParameterValue`:
- Concrete `np.ndarray` payloads behave exactly as before.
- Concrete `jax.Array` payloads round-trip through host memory via `np.asarray` (correct, just non-zero-copy).
- The remaining `jax.vmap` / tracer case is deferred — the demo runs the engine in a serial Python loop with one substitution per draw, which is fast enough for the prior-predictive pipeline.

If we later want true engine-level vmap, we'll re-open with a more focused proposal. Sorry for the noise.
